packages/x-data-grid/src/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -85,25 +86,37 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridRender
       }
     }, []);
 
+    const isSelectable = apiRef.current.isRowSelectable(id);
+
+    const checkboxPropsSelector = getCheckboxPropsSelector(
+      id,
+      rootProps.rowSelectionPropagation?.parents ?? false,
+    );
+    const { isIndeterminate, isChecked } = useGridSelector(apiRef, checkboxPropsSelector);
```

Comment:
What is the benefit of using a selector here if it's recreated every time?

Reply:
The selector is created every time yes, but due to the underlying ref used in `useGridSelector` it will be initialized only once and provide proper reactivity for the `isIndeterminate` and `isChecked` props.

The need to recreate the selector should also be solved when we shift to selectors with arguments (with v8)  as mentioned in https://github.com/mui/mui-x/pull/13757#discussion_r1784372973

Does it make sense?

Reply:
Ah, so `createSelector` is only used to avoid direct state access (reusing selectors we already have), but we don't benefit from its memoization (because we recreate the selector on every render).

How about using the v8 selector here?

Reply:
Yes, that will be the end goal when we enter the v8 pre-release phase. Currently, it's not possible as the parameters' order of [v7 selectors](https://github.com/mui/mui-x/blob/4148642b952dd4d4bfc70476bde2f29ec37bc905/packages/x-data-grid/src/utils/createSelector.ts#L105) and [v8 selectors](https://github.com/mui/mui-x/blob/4148642b952dd4d4bfc70476bde2f29ec37bc905/packages/x-data-grid/src/utils/createSelector.ts#L187) is different.
If we use the v8 selector here, we have to convert all the reused selectors to v8 ones for them to work properly.

Reply:
Ahh, you're right. The old selectors aren't compatible with v8 selectors 👍🏻

Reply:
Selectors are never reactively bound. There just wasn't any case where it made sense. For it to make sense we'd need a use-case like `const x = useGridSelector(api, condition ? selectorA : selectorB)`, but I don't see any case where the type of `x` would make sense. It would have also prevented `useGridSelector(api, state => state.x.y)`.